### PR TITLE
docs: Refactor API Kit Reference

### PR DESCRIPTION
**Pull request by germartinez**
## Context
This PR:
- Refactors and improves the API Kit technical reference

### CODE REVIEW

File: pages/reference-sdk-api-kit/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# API Kit Reference
+
+The [API Kit](https://github.com/safe-global/safe-core-sdk/tree/main/packages/api-kit) facilitates the interaction with the [Safe Transaction Service API](https://github.com/safe-global/safe-transaction-service).
```
Comment (tanay1337) [inline pages/reference-sdk-api-kit/overview.mdx:3]: Should we link to the docs pages here instead (for both links)?

**Review by tanay1337 — APPROVED** [review]
Thoughts on moving the "Quickstart" section from the Overview page to the Guides section?

File: pages/reference-sdk-api-kit/addsafedelegate.mdx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+```
+
+### `delegatorAddress`
+
+- **Type:** `string`
+
+The delegator address.
+
+```typescript focus=3
+const config = await apiKit.addSafeDelegate({
```
Comment (valle-xyz) [inline pages/reference-sdk-api-kit/addsafedelegate.mdx:87]: Add 
  safeAddress: '0x...',

here for consistency

Comment (valle-xyz) [reply pages/reference-sdk-api-kit/addsafedelegate.mdx:87]: Also at the other occurences.

File: pages/reference-sdk-api-kit/estimatesafetransaction.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  '0x...',
+  {
+    to: '0x...',
+    value: '0',
+    data: '0x'
+  }
+)
+```
+
+### `to`
```
Comment (valle-xyz) [inline pages/reference-sdk-api-kit/estimatesafetransaction.mdx:69]: I would add a differentiation, that this is sub data of the transaction data.

File: pages/reference-sdk-api-kit/proposetransaction.mdx [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+    gasToken: '0x...',
+    refundReceiver: '0x...',
+    nonce: '10'
+  },
+  senderAddress: '0x...',
+  senderSignature: '0x...'
+})
+```
+
+### `operation`
```
Comment (valle-xyz) [inline pages/reference-sdk-api-kit/proposetransaction.mdx:103]: This too should indicate that operation and the following parameters are part of the transaction data.

File: pages/reference-sdk-api-kit/proposetransaction.mdx [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    safeTxGas: '0x...',
+    baseGas: '123',
+    gasPrice: '123',
+    gasToken: '0x...',
+    refundReceiver: '0x...',
+    nonce: '10'
+  },
+  senderAddress: '0x...',
+  senderSignature: '0x...'
+})
```
Comment (valle-xyz) [inline pages/reference-sdk-api-kit/proposetransaction.mdx:124]: add origin //optional for consistency (also to the other calls).

File: pages/reference-sdk-api-kit/getalltransactions.mdx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+- **Type:** `boolean`
+
+If true, only executed transactions are returned.
+
+```typescript focus=4
+const allTxs = await apiKit.getAllTransactions(
+  safeAddress,
+  {
+    executed: true
```
Comment (valle-xyz) [inline pages/reference-sdk-api-kit/getalltransactions.mdx:73]: Weirdly enough, here I actually like that you omit the optional parameters as part of the transaction option object.

**Comment by valle-xyz** [comment]
Great PR!

* I (really) like that each function gets its own link in the sidebar. 
* I like that every input parameter is documented.
* Discussion: I am not super sure if it increases orientation that clicking on reference opens a new sidebar view. Viem and other docs have multiple layers of tabs in the sidebar and it works. What do you guys think? Would it be possible to create a small PR with a preview of that, so we can decice?
* The return parameter is not documented yet. I would say this is out of scope, but I would like to see it implemented. Should we open a ticket for that?

File: pages/reference-sdk-api-kit/addsafedelegate.mdx (hunk 2 above)
Comment (germartinez) [reply pages/reference-sdk-api-kit/addsafedelegate.mdx:87]: The `safeAddress` is optional, so to keep it clean I just used in its own section

File: pages/reference-sdk-api-kit/getalltransactions.mdx (hunk 6 above)
Comment (germartinez) [reply pages/reference-sdk-api-kit/getalltransactions.mdx:73]: I was doing that everywhere, just keeping the mandatory params + the optionals where we are in their own section

File: pages/reference-sdk-api-kit/proposetransaction.mdx (hunk 5 above)
Comment (germartinez) [reply pages/reference-sdk-api-kit/proposetransaction.mdx:124]: What I did everywhere:
- Add all the params, optionals included with the `// Optional ` comment in the Usage section.
- Avoid optional params as much as possible everywhere else.

File: pages/reference-sdk-api-kit/addsafedelegate.mdx (hunk 2 above)
Comment (valle-xyz) [reply pages/reference-sdk-api-kit/addsafedelegate.mdx:87]: Ok!

File: pages/reference-sdk-api-kit/proposetransaction.mdx (hunk 5 above)
Comment (valle-xyz) [reply pages/reference-sdk-api-kit/proposetransaction.mdx:124]: Ok, thanks for clarification.

File: pages/reference-sdk-api-kit/addmessagesignature.mdx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+The Safe message hash.
+
+```typescript focus=2
+await apiKit.addMessageSignature(
+  '0x...',
+  '0x...'
+)
+```
+
+### `siganture`
```
Comment (yagopv) [inline pages/reference-sdk-api-kit/addmessagesignature.mdx:50]:
```suggestion
### `signature`
```

**Review by valle-xyz — APPROVED** [review]
My last comments are still valid, but this looks good to me.

**Pull request merged**